packages/core/src/features/annotations/annotationsSlice.ts

```diff
@@ -42,17 +43,19 @@ const slice = createSlice({
   extraReducers: (builder) => {
     builder
       .addCase(fetchAnnotations.fulfilled, (state, action) => {
+        if (state.requestId != action.meta.requestId) return state;
```

Comment:
(This comment is not specific to this slice. Just placing it on the first one to allow for replies)

I see two potential issue with this pattern:
1. If any page makes multiple of the same request but with different request parameters, then only the last request will win.  Each reducer needs to be reviewed independently to see what it's doing.  If it completely replaces the state, then it should be ok.  If it's updated part of the state, especially if it's keyed based on the response, then it'll likely be problematic.
2. If the slice state is persisted to local storage, then we have the potential for any portal v2 tab to block the activity of any other portal v2 tab. While we do not officially support multiple tabs, this pattern may exacerbate any existing issues.

Reply:
My understanding is we always want the last request to win. All reducers in the PR except for two, which I will revert, completely replaces the state. 

None of these are persisted into local storage.

Reply:
I reverted GeneSymbol: which we should replace with a static file and ImageViewer as they both do partial update to the state.

Reply:
GeneSymbol has no issue with a race condition as it will replace a gene symbol with the same value. Which is why it should not need a slice or even be in the state.

Reply:
I took a look at the image viewer reducer, and I think there's a bug there.  Since it's doing an object spread of the existing data and the new data, it will accumulate key-value entries and never remove entries.  If I'm reading the code correctly, then it's a bug separate from this race-condition fix.  However, depending on how the bug is resolved, this race condition fix may need to be applied.